src/main/kotlin/io/sip3/salto/ce/sdp/SdpHandler.kt

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright 2018-2020 SIP3.IO, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.sip3.salto.ce.sdp
+
+import gov.nist.javax.sip.message.SIPMessage
+import gov.nist.javax.sip.message.SIPRequest
+import gov.nist.javax.sip.message.SIPResponse
+import io.sip3.commons.domain.Codec
+import io.sip3.commons.domain.SdpSession
+import io.sip3.salto.ce.RoutesCE
+import io.sip3.salto.ce.USE_LOCAL_CODEC
+import io.sip3.salto.ce.util.callId
+import io.sip3.salto.ce.util.sdpSessionId
+import io.sip3.salto.ce.util.sessionDescription
+import io.sip3.salto.ce.util.transactionId
+import io.vertx.core.AbstractVerticle
+import io.vertx.core.json.JsonObject
+import mu.KotlinLogging
+import org.restcomm.media.sdp.SessionDescription
+import org.restcomm.media.sdp.attributes.RtpMapAttribute
+import org.restcomm.media.sdp.fields.MediaDescriptionField
+
+class SdpHandler : AbstractVerticle() {
+
+    private val logger = KotlinLogging.logger {}
+
+    companion object {
+
+        const val CMD_UPDATE = "update"
+        const val CMD_TERMINATE = "terminate"
+    }
+
+    private var expirationDelay: Long = 1000
+    private var durationTimeout: Long = 3600000
+
+    private var codecs = mapOf<String, Codec>()
+    private val sessions = mutableMapOf<String, SdpSessionDescription>()
+
+    override fun start() {
+        config().getJsonObject("sip")?.getJsonObject("call")?.let { config ->
+            config.getLong("expiration-delay")?.let {
+                expirationDelay = it
+            }
+            config.getLong("duration-timeout")?.let {
+                durationTimeout = it
+            }
+        }
+
+        readCodecs(config())
+        vertx.eventBus().localConsumer<JsonObject>(RoutesCE.config_change) { event ->
+            try {
+                val config = event.body()
+                readCodecs(config)
+            } catch (e: Exception) {
+                logger.error("SdpHandler `readCodecs()` failed.", e)
+            }
+        }
+
+        vertx.setPeriodic(expirationDelay) {
+            terminateExpiredSessions()
+        }
+
+        vertx.eventBus().localConsumer<Pair<String, Any>>(RoutesCE.sdp_session) { event ->
+            try {
+                val (cmd, obj) = event.body()
+                when (cmd) {
+                    CMD_UPDATE -> handle(obj as SIPMessage)
```

Comment:
Let's call it `update()` in this case. In analogy with `terminate()`

Reply:
Let's simply rename `CMD_UPDATE` to `CMD_HANDLE` in this case.

Reply:
Ok